### DISCUSSION TITLE
Jeep 2018+: Counter/Checksum identification / Filtering by Byte

### DISCUSSION THREAD

**Discussion by jmccorm** [collin80/SavvyCAN#442] [Ideas]
On the 2018+ Jeep JL Wrangler (sharing an ECU with different late-model Jeep, Chrysler, Fiat, and Dodge vehicles), for many CAN message IDs, the last two bytes consist of a counter and a checksum.

It would be nice is SavvyCAN could automatically identify those components. It would also be nice if there was a way to filter our graphs such as this one to ignore those bytes (either automatically, or by allowing me to manually select individual bytes not to be graphed):
https://i.imgur.com/ROXhBpy.png

If you'd like to work with a sample of our vehicle's CAN bus data (as produced by Linux "candump"), you can find a copy at the bottom of this post in a file called "4mi-loop.zip":
https://www.jlwranglerforums.com/forum/threads/reverse-engineering-the-wranglers-can-bus-the-lasso-project.82139/page-3#post-1743660

It also includes a summary text file in human-readable format that gives you a basic idea of what the vehicle was doing at what timestamp.

We've identified a lot so far:
https://www.jlwranglerforums.com/forum/attachments/can-ids-2018-jl-wrangler-xlsx.547525/
...but it looks like this tool is going to allow us to identify a lot more. Everyone is welcome to use our own candump log file (zip file, above) to see what discoveries you can make for yourself!

Thanks,
Josh McCormick (jmccorm)

**Comment by collin80** [comment]
Make sure to try the pre-release / continuous builds. There hasn't been a release in a long time but the continuous builds happen every so often. They are now far ahead of the released version. The continuous build has support for turning off graphing of the data bytes you don't want to see in the Flow View. So, that feature already exists fortunately enough.

**Comment by jmccorm** [comment]
Thank you, that DOES help. I'm on the latest release (V207) and I've gone back to Flow View. I now see how to filter out bytes I don't want (thank you) although I'll probably have to puzzle out what it means to click on one of the squares and make it turn blue. HA!

Here's something that acts like a bug (and I have to remind you that I've got a 7 minute log file with 1033313 frames captured. It doesn't matter how short or long I make the slider for Chart Range, as I play back the log file, the Time Axis is jumping WILDLY between all sorts of intervals apparently ranging as high as 500. But each update is on a new horizontal axis. I'd send you a movie clip but I don't see an easy way to do it.

I have "Graph by timestamp" selected on the Flow View. On the main page, before I went into Flow View, I normalized frame timing, and then I clicked on Interpret Frames that didn't appear to do anything.

Hope this helps, but if not, let me know how better you'd like for me to collect and deliver data to you. And... BTW, very nice job on this software. I've done quite a lot of work without it, but now that I have that experience, I can easily see that this is going to help me fill in many of the gaps that still remain. WOW!

**Comment by collin80** [comment]
Just in case you didn't know, F1 does what you might expect of programs and
brings up help. The blue on the squares is in the help. Sometimes I don't
update the help files as often as I should and functionality won't be
covered. In this case, it is. What the blue means on numbers is to trigger
on that bit changing. For instance, if you click bit 20 then any time that
bit changes the playback will stop. You can use this to quickly seek around
and only stop when something you are investigating changes. It's like the
"Data seek values" which allow you to do the same thing but on a full byte.
Either way, they stop playback on a match. But, the bitfield going blue
stops whenever the bit changes, be that on or off.

I tried the "chart range" slider and it worked for me. It's a very new
addition from someone else and I had never tried it. But, it works great
for me, smoothly scaling the graph horizontal axis as I move it around. So,
if it is doing odd things that sounds like a bug. But, a strange one as I
was using your data and it worked for me.

Interpret frames on the main window will use loaded DBC files to interpret
the traffic. DBC files are a file format that describes signals in CAN
messages. You can set up signals that automatically interpret the data. For
instance, if you know where the RPM is sent you can turn the raw CAN data
into RPM and list the RPM. If you know how gear position is encoded you can
turn raw data into "DRIVE" "REVERSE" "NEUTRAL", etc. This makes viewing
data much more convenient. Basically, a DBC would be a way to store all the
things you've already decoded in a portable format. SavvyCAN can load DBC
files and it can write them as well. That's a whole other complication in a
program that I'm aware is already kind of complicated to use. But, being
able to store your work is pretty handy.

**Comment by jmccorm** [comment]
Collin,

I didn't consider that this had built-in help! Apologies, I'll try that on my next question. I'll circle back around with others and see if any of them want to standardize on SavvyCAN. Up until now, I've been using shell script (Raspbian / Linux) to do my translations. But I'm probably an outlier. Finally, I'll go back and play with my zooming issue. I just leave the zoom alone and it goes wild. If that continues, I'll capture hard data and try to file it as a bug report.
Thank you, by the way, for the verbose response. Totally appreciated!
Thanks,Josh MCCormick

    On Sunday, January 2, 2022, 05:15:28 PM CST, Collin Kidder ***@***.***> wrote:  
 
 

Just in case you didn't know, F1 does what you might expect of programs and
brings up help. The blue on the squares is in the help. Sometimes I don't
update the help files as often as I should and functionality won't be
covered. In this case, it is. What the blue means on numbers is to trigger
on that bit changing. For instance, if you click bit 20 then any time that
bit changes the playback will stop. You can use this to quickly seek around
and only stop when something you are investigating changes. It's like the
"Data seek values" which allow you to do the same thing but on a full byte.
Either way, they stop playback on a match. But, the bitfield going blue
stops whenever the bit changes, be that on or off.

I tried the "chart range" slider and it worked for me. It's a very new
addition from someone else and I had never tried it. But, it works great
for me, smoothly scaling the graph horizontal axis as I move it around. So,
if it is doing odd things that sounds like a bug. But, a strange one as I
was using your data and it worked for me.

Interpret frames on the main window will use loaded DBC files to interpret
the traffic. DBC files are a file format that describes signals in CAN
messages. You can set up signals that automatically interpret the data. For
instance, if you know where the RPM is sent you can turn the raw CAN data
into RPM and list the RPM. If you know how gear position is encoded you can
turn raw data into "DRIVE" "REVERSE" "NEUTRAL", etc. This makes viewing
data much more convenient. Basically, a DBC would be a way to store all the
things you've already decoded in a portable format. SavvyCAN can load DBC
files and it can write them as well. That's a whole other complication in a
program that I'm aware is already kind of complicated to use. But, being
able to store your work is pretty handy.



On Sat, Jan 1, 2022 at 10:22 PM jmccorm ***@***.***> wrote:

> Thank you, that DOES help. I'm on the latest release (V207) and I've gone
> back to Flow View. I now see how to filter out bytes I don't want (thank
> you) although I'll probably have to puzzle out what it means to click on
> one of the squares and make it turn blue. HA!
>
> Here's something that acts like a bug (and I have to remind you that I've
> got a 7 minute log file with 1033313 frames captured. It doesn't matter how
> short or long I make the slider for Chart Range, as I play back the log
> file, the Time Axis is jumping WILDLY between all sorts of intervals
> apparently ranging as high as 500. But each update is on a new horizontal
> axis. I'd send you a movie clip but I don't see an easy way to do it.
>
> I have "Graph by timestamp" selected on the Flow View. On the main page,
> before I went into Flow View, I normalized frame timing, and then I clicked
> on Interpret Frames that didn't appear to do anything.
>
> Hope this helps, but if not, let me know how better you'd like for me to
> collect and deliver data to you. And... BTW, very nice job on this
> software. I've done quite a lot of work without it, but now that I have
> that experience, I can easily see that this is going to help me fill in
> many of the gaps that still remain. WOW!
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/collin80/SavvyCAN/discussions/442#discussioncomment-1894853>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAQLWZPWGRWVFHWMT5SJUFDUT7AFTANCNFSM5LCXIKIQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
> You are receiving this because you commented.Message ID:
> ***@***.***>
>

—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by zoneh5** [comment]
I used the [Fiat ECU Remapping Tuning Files (Stage 1 & 2)](https://obd2technology.com/product/fiat-ecu-remapping-tuning-files-stage-1-2-for-enhanced-performance/) pack on my Punto—noticeable improvements with no added smoke, clean power curve. Great companion for WinOls users
